metaharvest.sh

```diff
@@ -15,154 +15,245 @@ source "${xpath}"/x.funx.sh
                 exit 0 # Success exit status
             ;;
-            -d | --download)
-                shift 1
-                eprintf "Downloading URLs for FASTQ data of '$1'"
-                # Also use sed to manage urls of paired-end read. 
-                _fetch_ena_project_json $1 | _extract_download_urls | \
-                    sed 's/;/\nwget -nc ftp:\/\//g'
-                exit 0
+            -e | --ena)
+                ena=true
+                shift
             ;;
-            -m | --metadata)
-                shift 1
-                eprintf "Fetching metadata of '$1'"
-                project_json=$(_fetch_ena_project_json $1)
-                geo_project_id=$(echo "${project_json}" \
-                    | jq -r '.[0] | .study_alias')
-
-                eprintf "Fetching GEO metadata of '${geo_project_id}'"
-                # To avoid an "argument too long" error, we need temporary files
-                # to save the metadata into.
-                geo_meta_file=$(mktemp)
-                ena_keys_file=$(mktemp)
-                
-                _fetch_series_file "${geo_project_id}" \
-                    | _series_to_csv > "${geo_meta_file}"
-                echo "${project_json}" \
-                    | _extract_geo_ena_sample_ids > "${ena_keys_file}"
-                "${xpath}/workers/fuse_csv.R" -c "geo_accession" \
-                    "${geo_meta_file}" "${ena_keys_file}"
-
-                rm "${geo_meta_file}"
-                rm "${ena_keys_file}"
-                exit 0
+            -g | --geo)
+                geo=true
+                shift
             ;;
+            -x* | --extra*)
+                # Test for '=' presence
+                rgx="^-x=|^--extra="
+                if [[ "$1" =~ $rgx ]]; then
+                    regular_entry="$1"
+                    regular_entry="${regular_entry#-x=}"
+                    regular_entry=",\"${regular_entry#--extra=}\""
+                    head_entry=",\"extra\""
+                    shift
+                else
+                    printf "Values need to be assigned to '--extra' option "
+                    printf "using the '=' operator.\n"
+                    printf "Use '--help' or '-h' to see the correct syntax.\n"
+                    exit 1 # Bad extra entry assignment
+                fi
+            ;;            
             *)
                 eprintf "Unrecognized option flag '$1'."
                 eprintf "Use '--help' or '-h' to see possible options."
-                exit 1 # Argument failure exit status: bad flag
+                exit 2 # Argument failure exit status: bad flag
             ;;
         esac
     else
-        eprintf "Bad argument '$1'.\n"
-        eprintf "Use '--help' or '-h' to see possible options."
-        exit 1 # Argument failure exit status: bad argument
+        # The first non-FRP sequence is taken as the ID argument
+        accession_id="$1"
+        break
```

Comment:
I did it because in my mind that argument was always supposed to be last--but I agree that there is no reason for that. Just replaced `break` with `shift` in commit 0297dcf49a07956dfd7528d092be080a745c4581.